customer/models.py

```diff
@@ -65,35 +66,46 @@ def create_or_update_reaction(message_id, related_id, route, to_number, sender_n
 
 
 class LoginOTP(models.Model):
-    customer = models.ForeignKey(Customer, on_delete=models.PROTECT)
+    customer = models.ForeignKey(Customer, on_delete=models.CASCADE, null=True, blank=True)
+    seller = models.ForeignKey(Tenant, on_delete=models.CASCADE, null=True, blank=True)
     otp_password = models.CharField(max_length=6)
     login_attempt = models.DecimalField(max_digits=1, decimal_places=0)
     generated_date = models.DateTimeField(null=False, default=datetime.now)
 
     @staticmethod
-    def get_otp(customer):
+    def get_otp(user, user_type):
         # Cleanup 24 hours old OTP
         time_threshold = datetime.now() - timedelta(hours=24)
         LoginOTP.objects.filter(generated_date__lt=time_threshold).delete()
 
         # Check if OPT exists, and attempt remaining
-        current_otp = LoginOTP.objects.filter(customer=customer).first()
+        if user_type == 'customer':
```

Comment:
combine the query using Q() filter depending on the conditions

Reply:
Updated